lib/ex_force/client/tesla/tesla.ex

```diff
@@ -41,20 +41,24 @@ defmodule ExForce.Client.Tesla do
   end
 
   def build_client(instance_url, opts) when is_binary(instance_url) do
-    Tesla.client(
-      [
-        {ExForce.Client.Tesla.Middleware,
-         {instance_url, Keyword.get(opts, :api_version, @default_api_version)}},
-        {Tesla.Middleware.Compression, format: "gzip"},
-        {Tesla.Middleware.JSON, engine: Jason},
-        {Tesla.Middleware.Headers, get_headers(opts)}
-      ],
-      Keyword.get(opts, :adapter)
-    )
+    custom_middleware = get_from_opts_or_config(opts, :middleware, [])
+
+    adapter = get_from_opts_or_config(opts, :adapter)
+
+    headers = get_headers(opts)
+
+    [
+      {ExForce.Client.Tesla.Middleware,
+       {instance_url, get_from_opts_or_config(opts, :api_version, @default_api_version)}},
+      {Tesla.Middleware.Compression, format: "gzip"},
+      {Tesla.Middleware.JSON, engine: Jason},
+      {Tesla.Middleware.Headers, headers},
+      custom_middleware
+    ]
+    |> List.flatten()
+    |> Tesla.client(adapter)
```

Comment:
Let's avoid using pipe here as we're not transforming a data here.

We can just `++` by enforcing passing list

```
  [
    {ExForce.Client.Tesla.Middleware,
      {instance_url, Keyword.get(opts, :api_version, @default_api_version)}}
    {Tesla.Middleware.Compression, format: "gzip"},
    {Tesla.Middleware.JSON, engine: Jason},
    {Tesla.Middleware.Headers, headers},
  ] ++ Keyword.get(opts, :append_middlewares, [])
```

Reply:
Sure - will update.

Reply:
Updated.